guides/development-workflow.md

```diff
@@ -0,0 +1,284 @@
+
+Doing this well requires a balanced approach to development.
+  * deliver gold plated solutions - we want to deliver the smallest amount of value as quickly as possible
+  * deliver too quickly - it may be at the expense of code quality
+
+#### The twenty-minute rules
+
+  1. being helpful is one of our values and this extends to development. If after spending 20 minutes on a problem you haven't made progress, invoke the twenty-minute rule with someone else and they will help
+
+  2. technical discussions that go on for twenty-minutes should invoke the second rule to provide a way to find agreement. Decide how much each of you cares out of 10 and work out if this discussion is worth continuing
+
+---
+
+## The lifecycle of a story
+
+### Start work on a new story
+
+1. assign yourself to the Trello card
+2. move the card to Trello's 'in progress' column
+3. make a new branch off of develop
+  ```
+  git fetch -a
+  git checkout develop
+  git branch <branch-name>
+  git checkout <branch-name>
+  ```
+4. meet the acceptance criteria
+5. [commit]() your code
+
+### Submit your story for code review
+
+1. push your branch to the remote repository
+  ```
+  git push origin <branch-name>
+  ```
+2. create a pull request into the `develop` branch:
+  * [GitHub](https://docs.gitlab.com/ee/gitlab-basics/add-merge-request.html)
+  * [GitLab](https://docs.gitlab.com/ee/gitlab-basics/add-merge-request.html)
+  * give it an [appropriate name and comment]()  
+3. Link the Trello card to the new pull request, linking directly or using the [power up](http://help.trello.com/article/1021-integrating-trello-with-zendesk)
+3. move the Trello card into the 'Awaiting column'
+
+### Reviewing another developer's pull request
+
+1. move the Trello card into the 'Under code review' column
+2. carry out a [code review]()
+  * if it's not ready:
+    - leave feedback
+    - mention the author so they're aware
+  * if it is ready:
+    - press the merge button
+    - delete the leftover branch
+    - move the Trello card into 'Deployed to Staging'
+
+---
+## The lifecycle of a ticket
+
+Tickets can fall into two types of work flow depending on their severity.
+
+If there an emergency that needs to be on fixed production as soon as possible? These are rare, but if so reach for a `hotfix` else a normal `fix`, which behaves just like a [story]().
+
+### Issue a hotfix
+  1. create a new card in Trello and move it to 'In progress'
+  2. create a new branch off of master
+  ```
+  git fetch -a
+  git checkout master
+  git pull
+  git branch <branch-name>
+  git checkout <branch-name>
+  ```
+  3. commit the fix
+  4. make a pull request into master
+  5. move the Trello card into 'Awaiting review'
+
+### Review and deploy a hotfix
+  1. review the code
+  2. check the fix is tested locally if possible
+  3. merge the pull request
```

Comment:
Put code onto production without testing it on staging first? We haven't done that for several years, and for good reason!